standalone/src/main.rs

```diff
@@ -212,28 +212,26 @@ fn main() {
                     let (backend, creg) = config.block_dev(block_dev);
 
                     let info = backend.info();
-                    let vioblk = hw::virtio::VirtioBlock::create(0x100, info);
+                    let vioblk = hw::virtio::PciVirtioBlock::new(0x100, info);
                     let id = inv
                         .register(&vioblk, format!("vioblk-{}", name), None)
                         .map_err(|e| -> std::io::Error { e.into() })?;
                     let _be_id = inv
                         .register_child(creg, id)
                         .map_err(|e| -> std::io::Error { e.into() })?;
 
-                    let blk =
-                        vioblk.inner_dev::<hw::virtio::block::VirtioBlock>();
-                    backend.attach(blk as Arc<dyn block::Device>, disp);
+                    backend
+                        .attach(vioblk.clone() as Arc<dyn block::Device>, disp);
 
                     chipset.pci_attach(bdf.unwrap(), vioblk);
                 }
                 "pci-virtio-viona" => {
                     let vnic_name =
                         dev.options.get("vnic").unwrap().as_str().unwrap();
 
-                    let viona = hw::virtio::viona::VirtioViona::create(
-                        vnic_name, 0x100, &hdl,
-                    )
-                    .unwrap();
+                    let viona =
+                        hw::virtio::PciVirtioViona::new(vnic_name, 0x100, &hdl)
+                            .unwrap();
```

Comment:
You can just replace this `.unwrap()` with `?`

Reply:
Yes indeed